### PR TITLE
add test for edf units param

### PR DESCRIPTION
**Pull request by agramfort**
follow up on https://github.com/mne-tools/mne-python/pull/11099#issuecomment-1229152014

cc @hoechenberger

### CODE REVIEW

File: mne/io/edf/edf.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def __init__(self, input_fname, eog=None, misc=None, stim_channel='auto',
             units = dict()
 
         for k, (this_ch, this_unit) in enumerate(orig_units.items()):
-            if this_unit != "" and this_unit in units:
+            if this_unit != "" and this_ch in units:
```
Comment (agramfort) [inline mne/io/edf/edf.py:150]: this was actually a bug ...

File: mne/io/edf/tests/test_edf.py [hunk 2]
```diff
@@ -77,6 +77,15 @@ def test_orig_units():
     assert orig_units['A1'] == 'µV'  # formerly 'uV' edit by _check_orig_units
 
 
+def test_units_params():
+    """Test enforcing original channel units."""
+
+    with pytest.raises(ValueError,
+                       match=r"Unit for channel .* is present .* Cannot "
+                       "overwrite it"):
+        raw = read_raw_edf(edf_path, units='V', preload=True)
```
Comment (hoechenberger) [inline mne/io/edf/tests/test_edf.py:86]:
```suggestion
       read_raw_edf(edf_path, units='V', preload=True)
```

File: mne/io/edf/tests/test_edf.py [hunk 3]
```diff
@@ -77,6 +77,15 @@ def test_orig_units():
     assert orig_units['A1'] == 'µV'  # formerly 'uV' edit by _check_orig_units
 
 
+def test_units_params():
+    """Test enforcing original channel units."""
+
```
Comment (hoechenberger) [inline mne/io/edf/tests/test_edf.py:82]:
```suggestion
    """Test enforcing original channel units."""
```

**Review by hoechenberger — CHANGES_REQUESTED** [review]
Style fixes, otherwise LGTM

**Review by hoechenberger — APPROVED** [review]

**Pull request merged**